lib/batchingReceiver.ts

```diff
@@ -0,0 +1,143 @@
+   * where the receiver should start receiving. Only one of offset, sequenceNumber, enqueuedTime, customFilter can be specified.
+   * `EventPosition.withCustomFilter()` should be used if you want more fine-grained control of the filtering.
+   * See https://github.com/Azure/amqpnetlite/wiki/Azure%20Service%20Bus%20Event%20Hubs for details.
+   */
+  constructor(context: ConnectionContext, partitionId: string | number, options?: ReceiveOptions) {
+    super(context, partitionId, options);
+  }
+
+  /**
+   * Receive a batch of EventData objects from an EventHub partition for a given count and a given max wait time in seconds, whichever
+   * happens first. This method can be used directly after creating the receiver object and **MUST NOT** be used along with the `start()` method.
+   *
+   * @param {number} maxMessageCount                 The maximum message count. Must be a value greater than 0.
+   * @param {number} [maxWaitTimeInSeconds]          The maximum wait time in seconds for which the Receiver should wait
+   * to receiver the said amount of messages. If not provided, it defaults to 60 seconds.
+   * @returns {Promise<EventData[]>} A promise that resolves with an array of EventData objects.
+   */
+  receive(maxMessageCount: number, maxWaitTimeInSeconds?: number): Promise<EventData[]> {
+    if (!maxMessageCount || (maxMessageCount && typeof maxMessageCount !== 'number')) {
+      throw new Error("'maxMessageCount' is a required parameter of type number with a value greater than 0.");
+    }
+
+    if (maxWaitTimeInSeconds == undefined) {
+      maxWaitTimeInSeconds = Constants.defaultOperationTimeoutInSeconds;
+    }
+
+    const eventDatas: EventData[] = [];
+    let timeOver = false;
+    return new Promise<EventData[]>((resolve, reject) => {
+      let onReceiveMessage: rheaPromise.OnAmqpEvent;
+      let onReceiveError: rheaPromise.OnAmqpEvent;
+      let waitTimer: any;
+      let actionAfterWaitTimeout: Func<void, void>;
+      // Final action to be performed after maxMessageCount is reached or the maxWaitTime is over.
+      const finalAction = (timeOver: boolean, data?: EventData) => {
+        // Resetting the mode. Now anyone can call start() or receive() again.
+        this._receiver.removeListener(Constants.receiverError, onReceiveError);
+        this._receiver.removeListener(Constants.message, onReceiveMessage);
+        if (!data) {
+          data = eventDatas.length ? eventDatas[eventDatas.length - 1] : undefined;
+        }
+        if (!timeOver) {
+          clearTimeout(waitTimer);
+        }
+        if (this.receiverRuntimeMetricEnabled && data) {
+          this.runtimeInfo.lastSequenceNumber = data.lastSequenceNumber;
+          this.runtimeInfo.lastEnqueuedTimeUtc = data.lastEnqueuedTime;
+          this.runtimeInfo.lastEnqueuedOffset = data.lastEnqueuedOffset;
+          this.runtimeInfo.retrievalTime = data.retrievalTime;
+        }
+        return this.close().then(() => { resolve(eventDatas); });
```

Comment:
it doesn't look like you handle the `.catch()` anywhere on this. make sure you add a handler for it (even if it's an empty function) to avoid the unhandled rejection warnings

Reply:
fwiw there's a [tslint rule](https://palantir.github.io/tslint/rules/no-floating-promises/) that might help catch these but i've never tried it